### ISSUE TITLE
withOutputParameterReturning does not work when using ignoreOtherParameters

### ISSUE THREAD

**Issue by jimpnet** [cpputest/cpputest#598]
The data specified by withOutputParameterReturning does not get copied into the buffer specified by withOutputParameter when used in conjunction with ignoreOtherParameters. It works fine if you don't ignore any parameters. This is in version 3.6.

**Comment by ryanplusplus** [comment]
@basvodde I'll take a look at this when I get a chance. Probably tomorrow local time.

**Comment by arstrube** [comment]
I ran the following test on the current head, and it passed:
```c++
TEST(MockSupportTest, outputParameterSucceedsWithIgnoreOtherParameters)
{
    int param = 1;
    int retval = 2;
    mock().expectOneCall("function")
          .withOutputParameterReturning("parameterName", &retval, sizeof(retval))
          .ignoreOtherParameters();
    mock().actualCall("function")
          .withOutputParameter("parameterName", &param).withParameter("boo", 5);
    CHECK_EQUAL(param, 2);
    CHECK_EQUAL(retval, 2);
    mock().checkExpectations();
    CHECK_NO_MOCK_FAILURE();
}
```

**Comment by jimpnet** [comment]
Does it pass in release 3.6?


Jim McClure
Senior Software Engineer - Embedded Systems

[cid:A00FCCDD-3B9C-4B2E-8F75-C109F0C49C88@hsd1.mn.comcast.net.]



4400 Baker Road, Minnetonka, MN 55343

TOLL FREE 888-346-3486  |  FAX 952-908-6129
www.peoplenetonline.com<http://www.peoplenetonline.com/>

PeopleNet is the leading provider of fleet mobility systems to the transportation industry, including truckload, LTL, private, and energy service fleets.

On Apr 14, 2015, at 12:52 PM, arstrube <notifications@github.com<mailto:notifications@github.com>> wrote:


I ran the following test on the current head, and it passed:

TEST(MockSupportTest, outputParameterSucceedsWithIgnoreOtherParameters)
{
    int param = 1;
    int retval = 2;
    mock().expectOneCall("function")
          .withOutputParameterReturning("parameterName", &retval, sizeof(retval))
          .ignoreOtherParameters();
    mock().actualCall("function")
          .withOutputParameter("parameterName", &param).withParameter("boo", 5);
    CHECK_EQUAL(param, 2);
    CHECK_EQUAL(retval, 2);
    mock().checkExpectations();
    CHECK_NO_MOCK_FAILURE();
}

**Comment by jimpnet** [comment]
I copy/pasted your test and it failed for me:

TEST(StRadioConfigSvc, StRadioConfigService_f)
{
    int param = 1;
    int retval = 2;
    mock().expectOneCall("function").withOutputParameterReturning("parameterName", &retval, sizeof(retval)).ignoreOtherParameters();
    mock().actualCall("function").withOutputParameter("parameterName", &param).withParameter("boo", 5);
    CHECK_EQUAL(param, 2);
    CHECK_EQUAL(retval, 2);
    mock().checkExpectations();
    CHECK_NO_MOCK_FAILURE();
}

Resulted in:

tests/stRadioConfigSvcTest.cpp:47: error: Failure in TEST(StRadioConfigSvc, StRadioConfigService_f)
        expected <1>
        but was  <2>
        difference starts at position 0 at: <          2         >

Jim McClure
Senior Software Engineer

PeopleNet

4400 Baker Road, Minnetonka, MN 55343

TOLL FREE 888.346.3486 | FAX 952.908.6129
www.peoplenetonline.com

PeopleNet is the leading provider of fleet mobility systems to the transportation industry, including truckload, LTL, private, and energy service fleets.
________________________________________

**Comment by arstrube** [comment]
3.6 didn't do me the honor of compiling straight away with Clang 6.0, so you beat me to it :D. Looks like this was a bug that has been fixed since.

**Comment by jimpnet** [comment]
Any idea when the next release will be?


Jim McClure
Senior Software Engineer - Embedded Systems

[cid:A00FCCDD-3B9C-4B2E-8F75-C109F0C49C88@hsd1.mn.comcast.net.]



4400 Baker Road, Minnetonka, MN 55343

TOLL FREE 888-346-3486  |  FAX 952-908-6129
www.peoplenetonline.com<http://www.peoplenetonline.com/>

PeopleNet is the leading provider of fleet mobility systems to the transportation industry, including truckload, LTL, private, and energy service fleets.

On Apr 14, 2015, at 1:06 PM, arstrube <notifications@github.com<mailto:notifications@github.com>> wrote:


3.6 didn't do me the honor of compiling straight away with Clang 6.0, so you beat me to it :D. Looks like this was a bug that has been fixed since.

**Comment by arstrube** [comment]
That would be for @basvodde to say.

**Comment by ryanplusplus** [comment]
Thanks @arstrube

**Comment by ryanplusplus** [comment]
See #362 for the fix

**Comment by basvodde** [comment]
@jimpnet Just released 3.7.1

**Comment by basvodde** [comment]
Fixed by PR. Closing
